routes/common/load_balancer_helper.rb

```diff
@@ -2,24 +2,49 @@
 
 class Routes::Common::LoadBalancerHelper < Routes::Common::Base
   def list
-    lbs = Serializers::LoadBalancer.serialize(project.load_balancers_dataset.authorized(@user.id, "LoadBalancer:view").all, {include_path: true})
-    @app.instance_variable_set(:@lbs, lbs)
-    @app.view "networking/load_balancer/index"
+    if @mode == AppMode::API
+      dataset = project.load_balancers_dataset
+      result = dataset.authorized(@user.id, "LoadBalancer:view").paginated_result(
+        start_after: @request.params["start_after"],
+        page_size: @request.params["page_size"],
+        order_column: @request.params["order_column"]
+      )
+
+      {
+        items: Serializers::LoadBalancer.serialize(result[:records]),
+        count: result[:count]
+      }
+    else
+      lbs = Serializers::LoadBalancer.serialize(project.load_balancers_dataset.authorized(@user.id, "LoadBalancer:view").all, {include_path: true})
+      @app.instance_variable_set(:@lbs, lbs)
+      @app.view "networking/load_balancer/index"
+    end
   end
 
   def post(name: nil)
     Authorization.authorize(@user.id, "LoadBalancer:create", project.id)
 
-    required_parameters = %w[name private_subnet_id algorithm src_port dst_port health_check_endpoint]
-    allowed_optional_parameters = %w[health_check_interval health_check_timeout health_check_up_threshold health_check_down_threshold]
+    required_parameters = %w[private_subnet_id src_port dst_port health_check_endpoint]
+    required_parameters << "name" if @mode == AppMode::WEB
+    allowed_optional_parameters = %w[algorithm health_check_interval health_check_timeout health_check_up_threshold health_check_down_threshold]
     request_body_params = Validation.validate_request_body(params, required_parameters, allowed_optional_parameters)
 
     ps = PrivateSubnet.from_ubid(request_body_params["private_subnet_id"])
+    unless ps
+      if @mode == AppMode::API
+        response.status = 404
+        @request.halt
+      else
+        flash["error"] = "Private subnet not found"
+        response.status = 404
+        @request.redirect "#{project.path}/load-balancer/create"
+      end
+    end
     Authorization.authorize(@user.id, "PrivateSubnet:view", ps.id)
 
     lb = Prog::Vnet::LoadBalancerNexus.assemble(
       ps.id,
-      name: name,
+      name: name || request_body_params["name"],
```

Comment:
Why do we need `request_body_params["name"]` here? We always get the name as a parameter, no?